### PR TITLE
Add logger and replace print calls with appropriate logging functions

### PR DESCRIPTION
**Pull request by kilacoda**
This commit adds a logger.py file and also replaces all `print` calls with logging functions like `logger.info`, `logger.warning`, `logger.error` etc.

Also adds rich as a dependency.

### CODE REVIEW

File: manim/__init__.py [hunk 1]
```diff
@@ -2,8 +2,7 @@
 from . import config
 from . import constants
 
-
-def main():
+def main():
```
Comment (leotrs) [inline manim/__init__.py:5]: A per PEP8, please leave two blank lines here

File: manim/config.py [hunk 2]
```diff
@@ -133,7 +135,7 @@ def parse_cli():
         )
         return parser.parse_args()
     except argparse.ArgumentError as err:
-        print(str(err))
+        logger.error(str(err))
```
Comment (leotrs) [inline manim/config.py:138]: Do we want to also emit an extra warning message, like on line 234?

File: manim/logger.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import logging
+from rich.logging import RichHandler
+
+## Setting config for logger.
```
Comment (leotrs) [inline manim/logger.py:4]: Not a very useful comment :) I would delete it

File: manim/mobject/svg/text_mobject.py [hunk 4]
```diff
@@ -287,7 +288,7 @@ def text2svg(self):
 
         if self.font == '':
             if NOT_SETTING_FONT_MSG != '':
-                print(NOT_SETTING_FONT_MSG)
+                logger.warning(NOT_SETTING_FONT_MSG)
```
Comment (leotrs) [inline manim/mobject/svg/text_mobject.py:291]: we may want to consider turning other error messages into constants as well.

File: manim/__init__.py (hunk 1 above)
Comment (eulertour) [reply manim/__init__.py:6]: but not two blank spaces!

File: manim/logger.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+logging.basicConfig(
+    level="NOTSET",
+    format="%(message)s",
```
Comment (eulertour) [inline manim/logger.py:7]: I actually think the filename could come in handy in logs. Maybe something that could be flag-configurable later.

File: manim/mobject/svg/text_mobject.py (hunk 4 above)
Comment (kilacoda) [reply manim/mobject/svg/text_mobject.py:291]: Agreed.

File: manim/config.py [hunk 6]
```diff
@@ -229,8 +231,8 @@ def get_camera_configuration(args):
         try:
             camera_config["background_color"] = colour.Color(args.color)
         except AttributeError as err:
-            print("Please use a valid color")
-            print(err)
+            logger.warning("Please use a valid color")
+            logger.error(err)
```
[the remark this answers is not in the document]
Comment (eulertour) [reply manim/config.py:235]: Or you could be fancy and use logger.exception(). Likewise inside the other exception blocks.

**Comment by huguesdevimeux** [comment]
Just a small thing, 
image

Could you something about the formatting of that? Because it looks ugly next to the great formatting of the logs.

**Pull request merged**